sqlparse/__init__.py

```diff
@@ -13,30 +13,71 @@
 from sqlparse import engine
 from sqlparse import filters
 from sqlparse import formatter
+from sqlparse import lexer
+from sqlparse.engine import grouping
 
 # Deprecated in 0.1.5. Will be removed in 0.2.0
 from sqlparse.exceptions import SQLParseError
 
 
-def parse(sql, encoding=None):
+def parse(
+    sql,
+    encoding=None,
+    pre_processes=None,
+    stmt_processes=None,
+    post_processes=None,
+    grouping_funcs=None
+):
     """Parse sql and return a list of statements.
 
     :param sql: A string containting one or more SQL statements.
     :param encoding: The encoding of the statement (optional).
+    :param pre_processes: pre_processes you want to run against parsed statements (optional).
+    :param stmt_processes: stmt_processes you want to run against parsed statements (optional).
+    :param post_processes: post_processes you want to run against parsed statements (optional).
+    :param grouping_funcs: grouping_funcs you want to run against parsed statements (optional).
     :returns: A tuple of :class:`~sqlparse.sql.Statement` instances.
     """
-    return tuple(parsestream(sql, encoding))
-
-
-def parsestream(stream, encoding=None):
+    stream = parse_stream(
+        sql,
+        encoding,
+        pre_processes=pre_processes or [],
+        stmt_processes=stmt_processes or [],
+        post_processes=post_processes or [],
+        grouping_funcs=grouping_funcs or []
+    )
+    # for statement in stream:
+    #     if statement.get_type() == 'CREATE':
+
+    return tuple(stream)
+
+
+def parse_stream(
+    stream,
+    encoding=None,
+    pre_processes=None,
+    stmt_processes=None,
+    post_processes=None,
+    grouping_funcs=None
+):
     """Parses sql statements from file-like object.
 
     :param stream: A file-like object.
     :param encoding: The encoding of the stream contents (optional).
+    :param pre_processes: pre_processes you want to run against parsed statements (optional).
+    :param stmt_processes: stmt_processes you want to run against parsed statements (optional).
+    :param post_processes: post_processes you want to run against parsed statements (optional).
+    :param grouping_funcs: grouping_funcs you want to run against parsed statements (optional).
     :returns: A generator of :class:`~sqlparse.sql.Statement` instances.
     """
-    stack = engine.FilterStack()
+    stack = engine.FilterStack(
+        pre_processes=pre_processes or [],
```

Comment:
This is OK to pass `pre_processes` directly to `FilterStack` and have `FilterStack` to handle `None` case.